src/crypto/CryptoInterface.c

```diff
@@ -81,57 +81,90 @@ void BSL_Crypto_SetRngGenerator(BSL_Crypto_RandBytesFn rand_gen_fn)
     rand_bytes_generator = rand_gen_fn;
 }
 
-int BSL_Crypto_UnwrapKey(BSL_Data_t *unwrapped_key_output, BSL_Data_t wrapped_key_plaintext, const char *key_id,
-                         size_t aes_variant)
+int BSL_Crypto_ClearKeyHandle(void *keyhandle)
 {
+    CHK_ARG_NONNULL(keyhandle);
+
+    BSLB_CryptoKey_t *key = (BSLB_CryptoKey_t *)keyhandle;
+    CHK_POSTCONDITION(BSL_SUCCESS == BSLB_CryptoKey_Deinit(key));
+    BSL_FREE(key);
+
+    return BSL_SUCCESS;
+}
+
+int BSL_Crypto_UnwrapKey(const void *kek_handle, size_t aes_variant, BSL_Data_t *wrapped_key, const void **cek_handle)
+{
+    BSLB_CryptoKey_t *kek = (BSLB_CryptoKey_t *)kek_handle;
+
+    BSLB_CryptoKey_t *cek = BSL_MALLOC(sizeof(BSLB_CryptoKey_t));
+
     const EVP_CIPHER *cipher = (aes_variant == BSL_CRYPTO_AES_128) ? EVP_aes_128_wrap() : EVP_aes_256_wrap();
     EVP_CIPHER_CTX   *ctx    = EVP_CIPHER_CTX_new();
-    ASSERT_ARG_NONNULL(ctx);
+    if (ctx == NULL)
+    {
+        BSL_FREE(cek);
+        return BSL_ERR_SECURITY_CONTEXT_CRYPTO_FAILED;
+    }
 
-    // Give the actual key extra margin on each side.
-    uint8_t  keybuf[128];
-    uint8_t *key = &keybuf[16];
-    memset(keybuf, 0, sizeof(keybuf));
+    BSL_Data_InitBuffer(&cek->raw, kek->raw.len);
```

Comment:
This is not always the case, but may be for RFC 9173 uses. It would be good to mark this as a sustainment issue that the CEK is limited to exactly the same length as the KEK and document that in the markup comment.